cmd/up.go

```diff
@@ -25,44 +27,61 @@ var upCmd = &cobra.Command{
 
 		c.DryRun, err = cmd.Flags().GetBool("dry-run")
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't get --dry-run flag: %v", err)
 		}
 
 		c.GcTag = GcTag
 
 		c.ClientPool, c.Discovery, err = restClientPool()
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't get Kubernetes client: %v", err)
+		}
+
+		// validate k8s version
+		version, err := utils.FetchVersion(c.Discovery)
+		if err != nil {
+			return fmt.Errorf("Can't verify Kubernetes version: %v", err)
+		}
+		if version.Major <= 1 && version.Minor < 7 {
+			return fmt.Errorf("Kubernetes with RBAC enabled (v1.7+) is required to run Kubeapps")
 		}
 
 		cwd, err := os.Getwd()
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't get current directory: %v", err)
 		}
 		wd := metadata.AbsPath(cwd)
 
 		manifest, err := fsGetFile("/kubeapps-ns.yaml")
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't read kubeapps manifest: %v", err)
 		}
+
 		objs, err := parseObjects(manifest)
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't parse kubeapps manifest: %v", err)
 		}
+
 		if err = c.Run(objs, wd); err != nil {
-			return err
+			return fmt.Errorf("Can't deploy kubeapps components: %v", err)
 		}
 
 		manifest, err = fsGetFile("/kubeapps-objs.yaml")
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't read kubeapps manifest: %v", err)
 		}
+
 		objs, err = parseObjects(manifest)
 		if err != nil {
-			return err
+			return fmt.Errorf("Can't parse kubeapps manifest: %v", err)
+		}
+
+		if err = c.Run(objs, wd); err != nil {
+			return fmt.Errorf("Can't install kubeapps components: %v", err)
 		}
 
-		return c.Run(objs, wd)
+		fmt.Println("Successfully installed kubeapps.")
```

Comment:
this gives the impression that `kubeapps` is now ready to be used (which is not true since it will be deploying for a while). I would set here a message that explains what to do next: monitor the deployment or wait for something.

Reply:
Ok so that probably should be directing user to run `$ kubectl get po --all-namespaces -w` for monitoring the deployment progress. WDYT ?

Reply:
maybe it is better to show the deployments?

Reply:
Working on a separate PR https://github.com/kubeapps/kubeapps/pull/52